google-cloud-clients/google-cloud-language/samples/src/main/java/com/google/cloud/examples/language/v1/LanguageSentimentText.java

```diff
@@ -0,0 +1,107 @@
+   * import com.google.cloud.language.v1.Document;
+   * import com.google.cloud.language.v1.EncodingType;
+   * import com.google.cloud.language.v1.LanguageServiceClient;
+   * import com.google.cloud.language.v1.Sentence;
+   */
+
+  /**
+   * Analyzing Sentiment in a String
+   *
+   * @param textContent The text content to analyze
+   */
+  public static void sampleAnalyzeSentiment(String textContent) {
+    try (LanguageServiceClient languageServiceClient = LanguageServiceClient.create()) {
+      // textContent = "I am so happy and joyful.";
+
+      // Available types: PLAIN_TEXT, HTML
+      Document.Type type = Document.Type.PLAIN_TEXT;
+
+      // Optional. If not specified, the language is automatically detected.
+      // For list of supported languages:
+      // https://cloud.google.com/natural-language/docs/languages
+      String language = "en";
+      Document document =
+          Document.newBuilder().setContent(textContent).setType(type).setLanguage(language).build();
+
+      // Available values: NONE, UTF8, UTF16, UTF32
+      EncodingType encodingType = EncodingType.UTF8;
+      AnalyzeSentimentRequest request =
+          AnalyzeSentimentRequest.newBuilder()
+              .setDocument(document)
+              .setEncodingType(encodingType)
+              .build();
+      AnalyzeSentimentResponse response = languageServiceClient.analyzeSentiment(request);
+      // Get overall sentiment of the input document
+      System.out.printf(
+          "Document sentiment score: %s\n", response.getDocumentSentiment().getScore());
+      System.out.printf(
+          "Document sentiment magnitude: %s\n", response.getDocumentSentiment().getMagnitude());
+      // Get sentiment for all sentences in the document
+      for (Sentence sentence : response.getSentencesList()) {
+        System.out.printf("Sentence text: %s\n", sentence.getText().getContent());
+        System.out.printf("Sentence sentiment score: %s\n", sentence.getSentiment().getScore());
+        System.out.printf(
+            "Sentence sentiment magnitude: %s\n", sentence.getSentiment().getMagnitude());
+      }
+      // Get the language of the text, which will be the same as
+      // the language specified in the request or, if not specified,
+      // the automatically-detected language.
+      System.out.printf("Language of the text: %s\n", response.getLanguage());
+    } catch (Exception exception) {
+      System.err.println("Failed to create the client due to: " + exception);
+    }
+  }
+  // [END language_sentiment_text]
+
+  public static void main(String[] args) throws Exception {
+    Options options = new Options();
+    options.addOption(
+        Option.builder("").required(false).hasArg(true).longOpt("text_content").build());
+
+    CommandLine cl = (new DefaultParser()).parse(options, args);
+    String textContent = cl.getOptionValue("text_content", "I am so happy and joyful.");
+
+    sampleAnalyzeSentiment(textContent);
+  }
```

Comment:
The nice thing about using tests to drive the samples is that you can have both positive and negative examples.